vault/namespaces_store.go

```diff
@@ -604,7 +666,13 @@ func (ns *NamespaceStore) ListNamespaceAccessors(ctx context.Context, includeRoo
 
 	entries := make([]string, 0, len(ns.namespaces))
 	for _, item := range ns.namespaces {
-		if !includeRoot && item.Namespace.ID == namespace.RootNamespaceID {
+		if !includeParent && item.Namespace.ID == parent.ID {
```

Comment:
Wonder if we can simplify the `List` behavior to take a filter to return the correct field rather than reimplementing it all again every time. 

Problem for later.

Reply:
Probably, but this is completely refactored in #1113 anyway.

Reply:
Ah, gotcha. Not yet looked at that one yet.